desktop-widgets/filterwidget2.cpp

```diff
@@ -0,0 +1,118 @@
+#include "desktop-widgets/filterwidget2.h"
+#include "desktop-widgets/simplewidgets.h"
+
+#include <QDoubleSpinBox>
+
+FilterWidget2::FilterWidget2(QWidget* parent)
+: QWidget(parent)
+, ui(new Ui::FilterWidget2())
+{
+	ui->setupUi(this);
+
+	FilterData data;
+	ui->minRating->setCurrentStars(data.minRating);
+	ui->maxRating->setCurrentStars(data.maxRating);
+	ui->minVisibility->setCurrentStars(data.minVisibility);
+	ui->maxVisibility->setCurrentStars(data.maxVisibility);
+	ui->minAirTemp->setValue(data.minAirTemp);
+	ui->maxAirTemp->setValue(data.maxAirTemp);
+	ui->minWaterTemp->setValue(data.minWaterTemp);
+	ui->maxWaterTemp->setValue(data.maxWaterTemp);
+
+	// TODO: unhide this when we discover how to search for equipment.
+	ui->equipment->hide();
+	ui->labelEquipment->hide();
+	ui->invertFilter->hide();
+
+	ui->to->setDate(data.to.date());
+
+	connect(ui->maxRating, &StarWidget::valueChanged,
+		[this]{ updateFilter(); }
```

Comment:
Personally, I dislike this kind of gratuitous lambdas. What's wrong with a pointer-to-member function? On the other hand I don't care much.

Reply:
Yes. See my comment two commits earlier. NAK on this. And I think this is the only reason to add C++14? Heck no.

Reply:
I would say that this is not gratuitous, to connect with a pointer to member I need to pass a method that receives the same arguments as the valueChanged, so I'd need to have a DateTimeSomethingChanged, StringSomethingChanged, DoubleSomethingChanged and IntSomethingChanged. while with lambdas I can basically ignore the type that's being send. I'm thinking on how to do this.

Reply:
it seems that I was wrong and I can now connect with different parameters, nice.

Reply:
Indeed I was confused about your explanation, because note how the lambda you defined above doesn't take a parameter either. So where would be the difference apart from the fact that one is a free-standing function and one is a member function...?